DotNet/Normalization/Application/Operations/Helpers/PropertyHelper.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Hl7.Fhir.FhirPath;
+using Hl7.Fhir.Introspection;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Utility;
+using Hl7.FhirPath;
+using Hl7.FhirPath.Expressions;
+using System.Reflection;
+using System.Text;
+
+namespace LantanaGroup.Link.Normalization.Application.Operations.Helpers
+{
+    public static class PropertyHelper
+    {
+        public static Base GetProperty(DomainResource resource, string targetFhirPath)
+        {
+            var found_property = resource.Select(targetFhirPath);
+
+            if (found_property.Any())
+            {
+                return found_property.First();
+            }
+
+            var symbolTable = new SymbolTable().AddStandardFP().AddFhirExtensions();
+            var compiler = new FhirPathCompiler(symbolTable);
+            var expression = compiler.Parse(targetFhirPath);
+            targetFhirPath = FilterFHIRPath(expression);
+
+            Base target = resource;
+
+            var resource_properties = ModelInfo.ModelInspector.FindClassMapping(resource.TypeName).PropertyMappings;
+
+            //Daniel - It looks like fhirpath works without including the resource name in the beginning. Removing this for now
+            //Strip out the first entry in the fhir path (the resource). We only need to traverse through the resource properties
+            //string target_properties_fhir_path = targetFhirPath.Substring(targetFhirPath.IndexOf(".") + 1);
+
+            PropertyMapping cur_property = null;
+
+            //Iterate through the fhir path to find if we have the entire fhirpath in memory. If not, contsruct it.
+            foreach (var property in targetFhirPath.Split('.'))
```

Comment:
It's unclear to me why this is needed. It looks like maybe it's to ensure that the hierarchy of the object model exists that the FHIRPath represents? So that it always returns a non-null value? Is that right? If so, why is that necessary?